### ISSUE TITLE
KiB suffix is not correctly parsed when constructing instances from string

### ISSUE THREAD

**Issue by saxbophone** [aphor/datasize#12]
Instances can be created with the `kiB` suffix but not the `KiB` suffix (this latter one is the correct capitalisation):

```python3
>>> from datasize import DataSize
>>>
>>> DataSize('10KiB')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.8/site-packages/datasize/__datasize__.py", line 112, in __new__
    raw_number = float(raw)
ValueError: could not convert string to float: '10Ki'
>>>
>>> DataSize('10kiB')
10240
```

**Comment by saxbophone** [comment]
@aphor I started doing some work on this in a fork, but I got a bit lost trying to work out how to tweak the unit tests properly. My progress on this can be found here: https://github.com/saxbophone/datasize?organization=saxbophone&organization=saxbophone

**Comment by aphor** [comment]
Thanks. I was waiting for this week to work on it, as I have some "staycation" time.

**Comment by saxbophone** [comment]
That is grand, thank you for the update.

**Comment by aphor** [comment]
Please have a look at https://github.com/aphor/datasize/tree/KiB and let me know if that is working for you, coverage is adequate, etc.

**Comment by saxbophone** [comment]
Thank you very much, I shall take a look and check it is working for me later this week when I have time.

**Comment by aphor** [comment]
Assuming everything is OK, I will merge it and cut a new rev/release to pypi when you do. Thanks!

**Comment by saxbophone** [comment]
Thank you for fixing this so promptly.

I have tested the amended version just now and it now works for my original use-case 👍 

Just a note, the incorrectly capitalised version (`kiB`) is now the one that raises an error, this doesn't concern me as I'm only interested in using the correct capitalisation, but I just wanted to check that this is intended behaviour?

```python3
>>> from datasize import DataSize
>>> DataSize('10KiB')
10240
>>> DataSize('10kiB')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/joshua/dev_build/datasize/datasize/__datasize__.py", line 133, in __new__
    raw_number = float(raw)
ValueError: could not convert string to float: '10ki'
>>>
```

**Comment by aphor** [comment]
I'll fix this today as well. Thanks for the testing. I'll bump the rev and push to PyPI when that's done, and close this ticket.

**Comment by saxbophone** [comment]
No problem, thanks again for fixing the issue so quickly. Looking forward to the new release.

**Comment by aphor** [comment]
Decided to do some refactoring, which I hope will make the formatter easier to debug. I added two new formatting options: m, and I, to autoformat either metric or IEC units, respectively, in case you want autoscaling units but want metric or IEC units exclusively.

**Comment by aphor** [comment]
merged, version bumped to 1.0.0

Released 1.0.0 to pypi!

**Comment by saxbophone** [comment]
Thank you very much!
